package/src/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -102,9 +182,36 @@ const renderImage = ({
     maxNumberOfFiles,
     numberOfAttachmentPickerImageColumns,
     selected,
+    setSelectedFiles,
     setSelectedImages,
   } = item;
-  const onPress = () => {
+
+  const contentType = lookup(asset.filename) || 'multipart/form-data';
+
+  const fileType = asset.filename
+    ? contentType.startsWith('image/')
+      ? 'image'
+      : 'video'
+    : asset.type === 'video'
+    ? 'video'
+    : 'image';
+
+  const videoDuration = asset.duration ? asset.duration : asset.playableDuration;
+
+  const ONE_HOUR_IN_SECONDS = 3600;
+
+  let duration = '00:00';
+
+  if (videoDuration) {
+    const isDurationLongerThanHour = videoDuration / ONE_HOUR_IN_SECONDS >= 1;
+    const formattedDurationParam = isDurationLongerThanHour ? 'HH:mm:ss' : 'mm:ss';
+    const formattedVideoDuration = dayjs
+      .duration(videoDuration, 'second')
+      .format(formattedDurationParam);
+    duration = formattedVideoDuration;
+  }
```

Comment:
I think we should move this logic to AttachmentVideo component. You can just pass the entire asset as prop to `AttachmentVideo` component.

Reply:
Fixed here - https://github.com/GetStream/stream-chat-react-native/pull/1332/files#diff-509e397cee3ad5197cf9a2710f06f9cbd3d2e586f6c23ea908bf95f3ab576161

Reply:
Fixed